src/Atmos/Model/precipitation.jl

```diff
@@ -92,7 +92,7 @@ function atmos_nodal_update_auxiliary_state!(
     q_rai = max(FT(0), state.ρq_rain * ρinv)
 
     # current state
-    ts = thermo_state(rain, state, aux)
+    ts = recover_thermo_state(rain, state, aux)
```

Comment:
Does this one assume that the `atmos_nodal_update_auxiliary_state` function from `moisture.jl` is called before the  `atmos_nodal_update_auxiliary_state` from `precipitation.jl`?

Reply:
Yes. This is why we call `atmos_nodal_update_auxiliary_state!(m.moisture, m, state, aux, t)` first in `AtmosModel` (so that all subsequent methods can call `recover_thermo_state`).